### PR TITLE
[main] Migrate to use hack/generate tool to generate images.yaml

### PR DESCRIPTION
**Pull request by pierDipi**
- Migrate to use hack/generate tool to generate images.yaml
- Related removal of unused scripts
- Overall cleanups

### CODE REVIEW

File: openshift/release/create-release-branch.sh [hunk 1]
```diff
@@ -18,8 +18,11 @@ git commit -sm ":fire: remove unneeded workflows" .github/
 # Copy the openshift extra files from the OPENSHIFT/main branch.
 git fetch openshift main
 git checkout openshift/main -- .github/workflows openshift OWNERS_ALIASES OWNERS Makefile
-make generate-dockerfiles
-make RELEASE=$release generate-release
+tag=${release/release-/}
+yq write --inplace openshift/project.yaml project.tag "knative-$tag"
+
+make generate-release
```
Comment (pierDipi) [inline openshift/release/create-release-branch.sh:24]: @creydr this is automatically changing the project.tag metadata when creating the branch

File: README.md [hunk 2]
```diff
@@ -63,50 +63,8 @@ For the CI setup, two repositories are of importance:
...
-After the file has been added to the folder as mentioned above, the
-job manifests itself will need to be generated as is described in the
-corresponding [ci-operator documentation](https://docs.google.com/document/d/1SQ_qlkcplqhe8h6ONXdgBr7YUVbs4oRSj4ISl3gpLW4/edit#heading=h.8w7nj9363nsd).
-
-Once all of this is done (Dockerfiles committed, ci-operator config
-created and job manifests generated) a PR must be opened against
-[openshift/release](https://github.com/openshift/releaseopenshift/release)
-to include all the ci-operator related files. Once
-this PR is merged, the CI setup for that branch is active.
+- [openshift-knative/hack](https://github.com/openshift-knative/hack) which
```
Comment (matzew) [inline README.md:66]: Do we have some sentence on how that works and is being used?

Comment (pierDipi) [reply README.md:66]: it's on the repository itself, do we need to duplicate that?

**Comment by pierDipi** [comment]
@creydr check the last commit I'm using this ability so we don't need the custom resolver:
https://github.com/openshift-knative/hack/blob/f614b3a209d0506f2d83e2edee3c7277f72e9748/cmd/generate/generate.go#L222-L224

File: README.md (hunk 2 above)
Comment (pierDipi) [reply README.md:66]: https://github.com/openshift-knative/hack#generate-openshiftrelease-config

File: openshift/ci-operator/Dockerfile.in [hunk 3]
```diff
@@ -1,11 +0,0 @@
-# Do not edit! This file was generated via Makefile
-FROM registry.ci.openshift.org/openshift/release:golang-1.18 as builder
-
-COPY . .
-RUN make install test-install
-
-FROM registry.access.redhat.com/ubi8/ubi-minimal
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply openshift/ci-operator/Dockerfile.in:7]: We use `registry.access.redhat.com/ubi8/ubi-minimal`, see https://github.com/openshift-knative/hack/blob/main/cmd/generate/Dockerfile.template

File: README.md (hunk 2 above)
Comment (matzew) [reply README.md:66]: Ok, that's enough info - thx

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**